adoc/chapters/programming_interface.adoc

```diff
@@ -3283,38 +3283,51 @@ describes them.
 .[apidef]#aspect::cpu#
 [role=synopsis,id=api:aspect-cpu]
 --
-A device that runs on a CPU.
-Devices with this aspect have device type [api]#info::device_type::cpu#.
+Indicates that the implementation identifies this device as a CPU that has
+device type [api]#info::device_type::cpu#.
+
+{note}A device with this aspect will typically share some or all of the
+execution resources available to the host {cpp} application.
+{endnote}
 --
 
 '''
 
 .[apidef]#aspect::gpu#
 [role=synopsis,id=api:aspect-gpu]
 --
-A device that can also be used to accelerate a 3D graphics API.
-Devices with this aspect have device type [api]#info::device_type::gpu#.
+Indicates that the implementation identifies this device as a GPU that has
+device type [api]#info::device_type::gpu#.
+
+{note}A device with this aspect may have additional capabilities for
+accelerating graphics operations, via SYCL image functionality and/or
+interoperability with graphics APIs.
+{endnote}
 --
 
 '''
 
 .[apidef]#aspect::accelerator#
 [role=synopsis,id=api:aspect-accelerator]
 --
-A dedicated accelerator device, usually using a peripheral interconnect for
-communication.
-Devices with this aspect have device type [api]#info::device_type::accelerator#.
+Indicates that the implementation identifies this device as an accelerator that
+has device type [api]#info::device_type::accelerator#.
+
+{note}A device with this aspect will typically be a dedicated accelerator
+device, with a peripheral interconnect for communication.
+{endnote}
```

Comment:
I think FPGA devices are normally exposed as this type.  Would a reader think that a "dedicated accelerator" might be an FPGA?  This is an honest question.  I don't know what most people would think of when reading the term "dedicated accelerator".

Reply:
Good question.  I have no idea, and I only used this phrasing because it was already there.

My reading of "accelerator" is something like "a piece of hardware that is not the host CPU", and I think "dedicated" implies that it's not going to be used to run anything except kernels you enqueue to it (i.e., it's not running anything related to the OS).  An FPGA would fit that description.  I guess an implementation might also choose to expose a "GPU" without graphics API support or one that isn't running a display as an "accelerator" instead of a "GPU".